source/birchwood/protocol/formatting.d

```diff
@@ -1,2 +1,99 @@
 module birchwood.protocol.formatting;
 
+import std.string;
+
+// Reset character; resets all formatting
+enum reset_code = '\x0F';
+
+// Toggle characters
+enum bold_code = '\x02';
+enum italic_code = '\x1D';
+enum underline_code = '\x1F';
+enum strikethrough_code = '\x1E';
+enum monospace_code = '\x11';
+enum reverse_colors_code = '\x16'; // NOT UNIVERSALLY SUPPORTED
+
+// Color characters
+enum ascii_color_code = '\x03';
+enum hex_color_code = '\x04';
+
+// Simple color codes
+enum simple_colors: string {
+    WHITE = "00",
+    BLACK = "01",
+    BLUE = "02",
+    GREEN = "03",
+    RED = "04",
+    BROWN = "05",
+    MAGENTA = "06",
+    ORANGE = "07",
+    YELLOW = "08",
+    LIGHT_GREEN = "09",
+    CYAN = "10",
+    LIGHT_CYAN = "11",
+    LIGHT_BLUE = "12",
+    PINK = "13",
+    GREY = "14",
+    LIGHT_GREY = "15",
+    DEFAULT = "99" // NOT UNIVERSALLY SUPPORTED
+}
+
+// Return the hex control character if color is a hexadecimal color code, the ASCII control character if color is two ASCII digits, and throw an exception if it's neither
+// This function might be useless now that set_fg and set_fg_bg have been changed, but I'll keep it in case it's needed later.
+char generate_color_control_char(string color) {
+    if (color.length == 6) {
+        return hex_color_code;
+    } else if (color.length == 2) {
+        return ascii_color_code;
+    } else {
+        throw new StringException("Invalid color code (must be either two ASCII digits or a hexadecimal code of the form RRGGBB)");
+    }
+}
+
+// Generates a string that changes the foreground color
+string set_foreground(string color) {
+    char[1] control_char;
+    if (color.length == 6) {
+        control_char[0] = hex_color_code;
+    } else if (color.length == 2) {
+        control_char[0] = ascii_color_code;
+    } else {
+        throw new StringException("Invalid color code (must be either two ASCII digits or a hexadecimal code of the form RRGGBB)");
+    }
+    return control_char ~ color;
+
+// Generate a string that sets the foreground and background color
+string set_foreground_background(string fg, string bg) {
```

Comment:
See argument above regarding using `enum` > `string` (directly)

Reply:
> See response above regarding hexadecimal color codes.

Noted.

I rate we keep what you have but then have some version of the function which takes in enum for a base set of predefined colors and simply calls this internal base function with said values.

Could you please add that?